src/toolkits/object_detection/object_detector.cpp

```diff
@@ -772,6 +772,62 @@ std::shared_ptr<MLModelWrapper> object_detector::export_to_coreml(
   return model_wrapper;
 }
 
+void object_detector::import_from_custom_model(variant_map_type model_data, size_t version){
+
+  auto it = model_data.find("_model");
+  flex_dict model = variant_get_value<flex_dict>(it->second);
+  model_data.erase(it);
+  auto it2 = model_data.find("_grid_shape");
+  std::vector<size_t> shape = variant_get_value<std::vector<size_t>>(it2->second);
+  model_data.insert(std::pair<std::string, size_t>("grid_height", shape[0]));
+  model_data.insert(std::pair<std::string,size_t>("grid_width",shape[1]));
+  model_data.insert(std::pair<std::string,std::string>("annotation_scale","pixel"));
+  model_data.insert(std::pair<std::string,std::string>("annotation_origin","top_left"));
+  model_data.insert(std::pair<std::string,std::string>("annotation_position","center"));
+  
+  state.clear();
```

Comment:
Just curious, why do we need to clear the state?

Reply:
I think it is a unity check say if you somehow import model twice :)